Framework/Core/include/Framework/ASoA.h

```diff
@@ -1988,6 +1990,75 @@ std::tuple<typename Cs::type...> getRowData(arrow::Table* table, T rowIterator,
 {
   return std::make_tuple(getSingleRowData<T, Cs>(table, rowIterator, ci, ai, globalIndex)...);
 }
+
+template <typename R, typename T, typename C>
+R getColumnValue(const T& rowIterator)
+{
+  return static_cast<R>(static_cast<C>(rowIterator).get());
+}
+
+template <typename R, typename T>
+using ColumnGetterFunction = R (*)(const T&);
+
+template <typename T, typename R>
+concept dynamic_with_common_getter = is_dynamic_v<T> &&
+                                     // lambda is callable without additional free args
+                                     framework::pack_size(typename T::bindings_t{}) == framework::pack_size(typename T::callable_t::args{}) &&
+                                     requires(T t) {
+                                       { t.get() } -> std::convertible_to<R>;
+                                     };
+
+template <typename T, typename R>
+concept persistent_with_common_getter = is_persistent_v<T> && requires(T t) {
+  { t.get() } -> std::convertible_to<R>;
+};
+
+template <typename R, typename T, persistent_with_common_getter<R> C>
+ColumnGetterFunction<R, T> createGetterPtr(const std::string_view& columnLabel)
+{
+  return columnLabel == C::columnLabel() ? &getColumnValue<R, T, C> : nullptr;
+}
+
+template <typename R, typename T, dynamic_with_common_getter<R> C>
+ColumnGetterFunction<R, T> createGetterPtr(const std::string_view& columnLabel)
+{
+  // allows user to use consistent formatting (with prefix) of all column labels
+  // by default there isn't 'f' prefix for dynamic column labels
+  bool isPrefixMatch = columnLabel.size() > 1 && columnLabel.substr(1) == C::columnLabel();
+  // check also exact match if user is aware of prefix missing
+  bool isExactMatch = columnLabel == C::columnLabel();
+
+  return (isPrefixMatch || isExactMatch) ? &getColumnValue<R, T, C> : nullptr;
```

Comment:
Looks much better, thanks. As a final note, why doing the comparison twice? Is it optimised out? How about:

```C++
if (columnLabel.size() > 1) {
  return columnLabel.substr(1) == C::columnLabel() ? &getColumnValue<R, T, C> : nullptr;
}
// The only remaining case is columnLabel.size() <= 1
return columnLabel.size() == 1 && columnLabel == C::columnLabel() : &getColumnValue<R, T, C> : nullptr;
```